### PR TITLE
Terraform 0.12 compatible

### PR DESCRIPTION
**Pull request by npalm**
Update the module to make it Terraform 0.12 compatible. Added a 0.12 example.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -38,7 +42,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 - Fix region defaults
 - Initial release, based on https://040code.github.io/2017/09/19/talk-immutable-infrastructure/
 
-[Unreleased]: https://github.com/philips-software/terraform-aws-vpc/compare/1.3.0...HEAD
+[Unreleased]: https://github.com/philips-software/terraform-aws-vpc/compare/1.4.0...HEAD
+[1.4.0]: https://github.com/philips-software/terraform-aws-vpc/compare/1.3.0...1.3.0
```
Comment (gertjanmaas) [inline CHANGELOG.md]:
```suggestion
[1.4.0]: https://github.com/philips-software/terraform-aws-vpc/compare/1.3.0...1.4.0
```

File: outputs.tf [hunk 2]
```diff
@@ -45,3 +45,8 @@ output "public_subnets_route_table" {
 output "private_subnets_route_table" {
   value = "${element(concat(aws_route_table.private_routetable.*.id, list("")), 0)}"
 }
+
+output "test" {
```
Comment (gertjanmaas) [inline outputs.tf]: Remove test code or rename to something more meaningful please :)

File: README.md [hunk 3]
```diff
@@ -50,31 +50,32 @@ module "y" {
...
-| vpc_id | ID of the VPC. |
+| availability\_zones | List of the availability zones. |
+| nat\_gateway\_public\_ip | Public IP address of the NAT gateway. |
+| private\_dns\_zone\_id | ID of the the private DNS zone, optional. |
+| private\_domain\_name | Private domain name, optional. |
+| private\_subnets | List of the private subnets. |
+| private\_subnets\_route\_table |  |
+| public\_subnets | List of the public subnets. |
+| public\_subnets\_route\_table |  |
+| test |  |
```
Comment (gertjanmaas) [inline README.md]: Remove test code or name to something meaningful please

File: main.tf [hunk 4]
```diff
@@ -167,17 +167,30 @@ data "aws_vpc_endpoint_service" "s3" {
 }
 
 resource "aws_vpc_endpoint" "s3_vpc_endpoint" {
-  count           = "${var.create_s3_vpc_endpoint ? 1 : 0}"
-  vpc_id          = "${aws_vpc.vpc.id}"
-  service_name    = "${data.aws_vpc_endpoint_service.s3.service_name}"
-  route_table_ids = ["${concat(aws_route_table.public_routetable.*.id, aws_route_table.private_routetable.*.id)}"]
```
Comment (gertjanmaas) [inline main.tf]: Why is the `route_table_ids` removed?

**Review by gertjanmaas — CHANGES_REQUESTED** [review]
Looks good, couple of small things to fix.

File: README.md (hunk 3 above)
Comment (npalm) [reply README.md]: Fixed

File: outputs.tf (hunk 2 above)
Comment (npalm) [reply outputs.tf]: fixed

**Comment by npalm** [comment]
@gertjanmaas thanks for the comments, issues should now be resolved

**Review by gertjanmaas — APPROVED** [review]

**Pull request merged**